chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java

```diff
@@ -125,7 +125,8 @@ public void initDB() {
         openDatabase(options);
         alive = true;
       } catch (IOException ioe) {

```

Comment:
`IOException ioe ` in there : Will there be any other types of errors here, all about database corruption?

Reply:
According to your analysis, a difference should be made.

Reply:
> InvalidPathException
So what difference should be made? When a NativeDB.DBException means database corruption and an  IOException should be thrown? @halibobo1205

Reply:
Sorry, I guess I wasn't clear enough, you may get ideas here #5223

Reply:
> #5223

That seems to be more precise, Just use the requirement "if (e.getMessage().contains("Corruption:")) {" should find the corruption certainly.